test/DynamoCoreTests/Logging/AnalyticsServiceTest.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Dynamo.Configuration;
+using Dynamo.Graph.Nodes;
+using Dynamo.Graph.Nodes.ZeroTouch;
+using Dynamo.Graph.Workspaces;
+using Dynamo.Interfaces;
+using Dynamo.Models;
+using Dynamo.Scheduler;
+using DynamoShapeManager;
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using SystemTestServices;
+using TestServices;
+
+namespace Dynamo.Tests.Loggings
+{
+    [TestFixture]
+    class AnalyticsServiceTest : DynamoModelTestBase
+    {
+        //We need to override this function because the one in DynamoModelTestBase is setting StartInTestMode = true
+        protected override DynamoModel.IStartConfiguration CreateStartConfiguration(IPreferences settings)
+        {
+            return new DynamoModel.DefaultStartConfiguration()
+            {
+                PathResolver = pathResolver,
+                StartInTestMode = false,
+                GeometryFactoryPath = preloader.GeometryFactoryPath,
+                Preferences = settings,
+                ProcessMode = TaskProcessMode.Synchronous
+            };
+        }
+
+        /// <summary>
+        /// This test method will validate that the AnalyticsService.OnWorkspaceAdded (CustomNodeWorkspaceModel) is executed
+        /// </summary>
+        [Test]
+        [Category("UnitTests")]
+        public void TestOnWorkspaceAdded()
+        {
+            //Arrange
+            // Open/Run XML test graph
+            string openPath = Path.Combine(TestDirectory, @"core\Angle.dyn");
+            RunModel(openPath);
+            int InitialNodesCount = CurrentDynamoModel.CurrentWorkspace.Nodes.Count();
+
+            // Convert a DSFunction node Line.ByPointDirectionLength to custom node.
+            var workspace = CurrentDynamoModel.CurrentWorkspace;
+            var node = workspace.Nodes.OfType<DSFunction>().First();
+
+            List<NodeModel> selectionSet = new List<NodeModel>() { node };
+            var customWorkspace = CurrentDynamoModel.CustomNodeManager.Collapse(
+                selectionSet.AsEnumerable(),
+                Enumerable.Empty<Dynamo.Graph.Notes.NoteModel>(),
+                CurrentDynamoModel.CurrentWorkspace,
+                true,
+                new FunctionNamePromptEventArgs
+                {
+                    Category = "Testing",
+                    Description = "",
+                    Name = "__AnalyticsServiceTest__",
+                    Success = true
+                }) as CustomNodeWorkspaceModel;
+
+            //Act
+            //This will execute the 
```

Comment:
Same here incomplete comment

Reply:
@QilongTang 
Fixed, I completed the comment in the commit:
https://github.com/DynamoDS/Dynamo/pull/10546/commits/001c77502c11c8979b14f6d879577ba79a2eaf37